misc/multiaddr/src/lib.rs

```diff
@@ -303,6 +303,44 @@ impl TryFrom<Vec<u8>> for Multiaddr {
     }
 }
 
+/// Create a [`Multiaddr`] from the given IP address and suffix.
+pub fn ip_to_multiaddr(ip: IpAddr, suffix: &[Protocol]) -> Multiaddr {
```

Comment:
This is the same as `Multiaddr::from(ip).with(suffix)` (assuming there's only one suffix, which is the case everywhere this function is called).
I know it's not you who wrote this function, but I'd be in favour of removing it.

Reply:
This actually isn’t the case ― `host_addresses` calls it with multiple suffixes.

Reply:
Removed it anyway.